executor/cook/progress.py

```diff
@@ -150,16 +150,19 @@ def tail(self, sleep_time_ms):
 
             logging.info('{} has been created, reading contents'.format(self.target_file))
             target_file_obj = open(self.target_file, 'r')
+            line_index = 0
             while not self.stop_signal.isSet():
                 line = target_file_obj.readline(self.max_bytes_read_per_line)
                 if not line:
                     # exit if program has completed and there are no more lines to read
                     if self.task_completed_signal.isSet():
-                        logging.info('Done processing file for progress messages')
+                        cio.print_and_log('Done processing progress messages, {} lines read'.format(
+                            line_index))
                         break
                     # no new line available, sleep before trying again
                     time.sleep(sleep_param)
                     continue
+                line_index += 1
```

Comment:
We're assuming that `readline` returns a whole line, which may not be true.

Reply:
Added a check for `line.endswith(os.linesep)`